Search.setIndex({"docnames": ["index", "installation", "modules", "pyseqrna", "usage"], "filenames": ["index.rst", "installation.rst", "modules.rst", "pyseqrna.rst", "usage.rst"], "titles": ["pySeqRNA: an automated Python package for RNASeq data analysis", "Installation", "Modules", "Introduction", "Usage"], "terms": {"author": [0, 3], "naveen": [0, 3], "duhan": [0, 3], "date": 0, "oct": 0, "09": 0, "2022": [0, 3], "version": [0, 2], "latest": 0, "i": [0, 1, 3], "todai": 0, "massiv": 0, "amount": 0, "ar": [0, 3], "gener": [0, 3], "next": 0, "sequenc": 0, "ng": 0, "technologi": 0, "In": 0, "recent": 0, "year": 0, "mani": 0, "algorithm": 0, "statist": [0, 3], "method": 0, "softwar": 0, "tool": [0, 3], "have": 0, "been": 0, "develop": 0, "perform": [0, 3], "individu": 0, "step": 0, "variou": 0, "applic": 0, "howev": 0, "streamlin": 0, "remain": 0, "signific": 0, "barrier": 0, "effect": 0, "util": [0, 2], "we": 0, "allow": 0, "fast": 0, "effici": 0, "manag": 0, "reproduc": 0, "rna": [0, 3], "seq": 0, "It": [0, 3], "us": [0, 1, 3], "current": [0, 3], "newli": 0, "written": 0, "script": [0, 3], "without": [0, 3], "confin": 0, "user": 0, "collect": 0, "pre": 0, "defin": 0, "environ": 0, "combin": [0, 3], "command": [0, 3], "line": 0, "custom": 0, "keyword": 0, "transcriptom": 0, "differenti": [0, 2], "gene": [0, 2], "express": [0, 3], "usag": 0, "real": 0, "provid": [0, 3], "instal": 0, "modul": 0, "qualiti": [0, 2], "submodul": [0, 2], "align": [0, 2], "quantif": [0, 2], "function": [0, 2], "annot": [0, 2], "index": [0, 3], "search": [0, 3], "page": 0, "pyseqrna": [1, 2], "avail": [1, 3], "pypi": 1, "could": 1, "via": 1, "pip": 1, "To": [1, 3], "from": [1, 3], "sourc": 1, "clone": 1, "git": 1, "repo": 1, "http": [1, 3], "github": [1, 3], "com": [1, 3], "navduhan": 1, "Then": 1, "r": [1, 3], "requir": 1, "txt": 1, "e": [1, 3], "introduct": 2, "quality_check": 2, "fastqcrun": [2, 3], "quality_trim": 2, "flexbarrun": [2, 3], "trim_galorerun": [2, 3], "trimmomaticrun": [2, 3], "ribosom": 2, "sortmernarun": [2, 3], "star_align": [2, 3], "build_index": [2, 3], "check_index": [2, 3], "run_align": [2, 3], "hisat2_align": [2, 3], "pyseqrna_stat": 2, "align_stat": [2, 3], "featurecount": [2, 3], "htseqcount": [2, 3], "multimapped_group": 2, "countmmg": [2, 3], "normalize_count": 2, "normal": [2, 3], "cpm": [2, 3], "rpkm": [2, 3], "tpm": [2, 3], "meanratiocount": [2, 3], "boxplot": [2, 3], "cluster": 2, "clustersampl": [2, 3], "leaf_label": [2, 3], "differential_express": 2, "gene_descript": [2, 3], "add_nam": [2, 3], "add_names_annot": [2, 3], "degfilt": [2, 3], "rundeseq2": [2, 3], "run_edg": [2, 3], "gene_ontologi": 2, "geneontologi": [2, 3], "barplotgo": [2, 3], "dotplotgo": [2, 3], "enrichgo": [2, 3], "pathwai": 2, "barplotkegg": [2, 3], "dotplotkegg": [2, 3], "enrichkegg": [2, 3], "arg_pars": 2, "pyseqrna_util": 2, "pyseqrnalogg": [2, 3], "change_attribut": [2, 3], "change_id": [2, 3], "check_fil": [2, 3], "check_path": [2, 3], "check_statu": [2, 3], "clusterrun": [2, 3], "findfil": [2, 3], "getfil": [2, 3], "getgen": [2, 3], "get_basenam": [2, 3], "get_cpu": [2, 3], "get_directori": [2, 3], "get_file_extens": [2, 3], "get_par": [2, 3], "make_directori": [2, 3], "parse_config_fil": [2, 3], "parse_gff": [2, 3], "read_input_fil": [2, 3], "replace_cpu": [2, 3], "pyseqrna_plot": 2, "plotheatmap": [2, 3], "plotma": [2, 3], "plotvenn": [2, 3], "plotvolcano": [2, 3], "read": 3, "configur": 3, "titl": 3, "thi": 3, "contain": 3, "check": 3, "creat": 3, "mai": 3, "19": 3, "2021": 3, "sampledict": 3, "none": 3, "configfil": 3, "slurm": 3, "fals": 3, "mem": 3, "10": 3, "cpu": 3, "8": 3, "task": 3, "1": 3, "pairedend": 3, "aftertrim": 3, "outdir": 3, "dep": 3, "fastqc": 3, "www": 3, "bioinformat": 3, "babraham": 3, "ac": 3, "uk": 3, "project": 3, "paramet": 3, "sampl": 3, "dictionari": 3, "file": 3, "default": 3, "param": 3, "true": 3, "enabl": 3, "job": 3, "schedul": 3, "hpc": 3, "memori": 3, "number": 3, "thread": 3, "per": 3, "pair": 3, "after": 3, "trim": 3, "output": 3, "directori": 3, "result": 3, "work": 3, "depend": 3, "20": 3, "adapt": 3, "base": 3, "flexbar": 3, "seqan": 3, "samplesdict": 3, "A": 3, "inform": 3, "config": 3, "ini": 3, "gb": 3, "run": 3, "total": 3, "id": 3, "which": 3, "trmmomat": 3, "sortmerna": 3, "remov": 3, "juli": 3, "31": 3, "rnadatabas": 3, "execut": 3, "fastq": 3, "present": 3, "end": 3, "other": 3, "class": 3, "21": 3, "genom": 3, "object": 3, "star": 3, "program": 3, "path": 3, "get": 3, "gff": 3, "build": 3, "geneom": 3, "featur": 3, "valid": 3, "exist": 3, "return": 3, "target": 3, "against": 3, "refer": 3, "pyseqrna_result": 3, "hisat2": 3, "largeindex": 3, "larg": 3, "spetemb": 3, "trimdict": 3, "bamdict": 3, "ribodict": 3, "calcul": 3, "raw": 3, "all": 3, "trindict": 3, "bam": 3, "filter": 3, "datafram": 3, "type": 3, "conta": 3, "count": 3, "29": 3, "paramt": 3, "gtf": 3, "htseq": 3, "multimap": 3, "group": 3, "june": 3, "5": 3, "mincount": 3, "100": 3, "percentsampl": 3, "0": 3, "minimum": 3, "percent": 3, "convert": 3, "august": 3, "3": 3, "countfil": 3, "featurefil": 3, "typefil": 3, "keytyp": 3, "ncbi": 3, "attribut": 3, "genecolumn": 3, "plot": 3, "figsiz": 3, "million": 3, "log": 3, "figur": 3, "size": 3, "killobas": 3, "medianratio": 3, "data": 3, "counttyp": 3, "kwarg": 3, "make": 3, "box": 3, "color": 3, "accord": 3, "thei": 3, "belong": 3, "list": 3, "column": 3, "option": 3, "argument": 3, "similar": 3, "dendrogram": 3, "between": 3, "2": 3, "countdf": 3, "temp": 3, "funtion": 3, "leaf": 3, "label": 3, "leav": 3, "arrang": 3, "find": 3, "octob": 3, "22": 3, "speci": 3, "degfil": 3, "fetch": 3, "name": 3, "descript": 3, "add": 3, "deg": 3, "degdf": 3, "comparelist": 3, "fdr": 3, "05": 3, "fold": 3, "6": 3, "replic": 3, "extracolumn": 3, "given": 3, "differanti": 3, "comparison": 3, "discoveri": 3, "rate": 3, "chang": 3, "valu": 3, "The": 3, "log2": 3, "want": 3, "barplot": 3, "targetfil": 3, "design": 3, "gene_column": 3, "mmg": 3, "subset": 3, "lib": 3, "wrapper": 3, "deseq2": 3, "packag": 3, "differeanti": 3, "analysi": 3, "tab": 3, "delimit": 3, "contan": 3, "compar": 3, "first": 3, "If": 3, "librari": 3, "bcv": 3, "4": 3, "edger": 3, "biolog": 3, "coeffici": 3, "variat": 3, "ontologi": 3, "enrich": 3, "januari": 3, "ensembl": 3, "taxid": 3, "ex": 3, "arabidopsi": 3, "thaliana": 3, "athaliana": 3, "plant": 3, "anim": 3, "taxonomi": 3, "df": 3, "nrow": 3, "colorbi": 3, "logpvalu": 3, "row": 3, "bar": 3, "dotplot": 3, "dot": 3, "pvaluecutoff": 3, "plottyp": 3, "p": 3, "cutoff": 3, "need": 3, "visual": 3, "rtype": 3, "kegg": 3, "parser": 3, "11": 3, "mode": 3, "intial": 3, "logger": 3, "arg": 3, "synonym": 3, "replac": 3, "retrun": 3, "onli": 3, "job_id": 3, "statu": 3, "complet": 3, "job_nam": 3, "sout": 3, "serror": 3, "time": 3, "submit": 3, "excut": 3, "allot": 3, "sbatch": 3, "searchpath": 3, "searchpattern": 3, "recurs": 3, "verbos": 3, "pattern": 3, "print": 3, "patteren": 3, "direcori": 3, "multisheet": 3, "genetyp": 3, "extract": 3, "differential": 3, "up": 3, "down": 3, "filepath": 3, "full": 3, "actual": 3, "system": 3, "integ": 3, "int": 3, "80": 3, "extens": 3, "dir": 3, "infil": 3, "pars": 3, "specif": 3, "inpath": 3, "input": 3, "also": 3, "possibl": 3, "inputfil": 3, "infrom": 3, "about": 3, "inputpath": 3, "args2": 3, "septemb": 3, "15": 3, "novemb": 3, "num": 3, "50": 3, "figdim": 3, "12": 3, "heatmap": 3, "25": 3, "dimens": 3, "comp": 3, "red": 3, "grei": 3, "green": 3, "dim": 3, "dotsiz": 3, "markertyp": 3, "o": 3, "alpha": 3, "ma": 3, "marketyp": 3, "shape": 3, "transpar": 3, "deglabel": 3, "fontsiz": 3, "14": 3, "dpi": 3, "300": 3, "venn": 3, "diagram": 3, "excel": 3, "sheet": 3, "wise": 3, "how": 3, "put": 3, "labe": 3, "either": 3, "font": 3, "resolut": 3, "pvalu": 3, "volcano": 3}, "objects": {"": [[3, 0, 0, "-", "pyseqrna"]], "pyseqrna": [[3, 0, 0, "-", "aligners"], [3, 0, 0, "-", "arg_parser"], [3, 0, 0, "-", "clustering"], [3, 0, 0, "-", "differential_expression"], [3, 0, 0, "-", "gene_ontology"], [3, 0, 0, "-", "multimapped_groups"], [3, 0, 0, "-", "normalize_counts"], [3, 0, 0, "-", "pathway"], [3, 0, 0, "-", "pyseqrna_plots"], [3, 0, 0, "-", "pyseqrna_stats"], [3, 0, 0, "-", "pyseqrna_utils"], [3, 0, 0, "-", "quality_check"], [3, 0, 0, "-", "quality_trimming"], [3, 0, 0, "-", "quantification"], [3, 0, 0, "-", "ribosomal"], [3, 0, 0, "-", "version"]], "pyseqrna.aligners": [[3, 1, 1, "", "STAR_Aligner"], [3, 1, 1, "", "hisat2_Aligner"]], "pyseqrna.aligners.STAR_Aligner": [[3, 2, 1, "", "build_index"], [3, 2, 1, "", "check_index"], [3, 2, 1, "", "run_Alignment"]], "pyseqrna.aligners.hisat2_Aligner": [[3, 2, 1, "", "build_index"], [3, 2, 1, "", "check_index"], [3, 2, 1, "", "run_Alignment"]], "pyseqrna.clustering": [[3, 3, 1, "", "clusterSample"], [3, 3, 1, "", "leaf_label"]], "pyseqrna.differential_expression": [[3, 1, 1, "", "Gene_Description"], [3, 3, 1, "", "degFilter"], [3, 3, 1, "", "runDESeq2"], [3, 3, 1, "", "run_edgeR"]], "pyseqrna.differential_expression.Gene_Description": [[3, 2, 1, "", "add_names"], [3, 2, 1, "", "add_names_annotation"]], "pyseqrna.gene_ontology": [[3, 1, 1, "", "GeneOntology"]], "pyseqrna.gene_ontology.GeneOntology": [[3, 2, 1, "", "barplotGO"], [3, 2, 1, "", "dotplotGO"], [3, 2, 1, "", "enrichGO"]], "pyseqrna.multimapped_groups": [[3, 3, 1, "", "countMMG"]], "pyseqrna.normalize_counts": [[3, 1, 1, "", "Normalization"], [3, 3, 1, "", "boxplot"]], "pyseqrna.normalize_counts.Normalization": [[3, 2, 1, "", "CPM"], [3, 2, 1, "", "RPKM"], [3, 2, 1, "", "TPM"], [3, 2, 1, "", "meanRatioCount"]], "pyseqrna.pathway": [[3, 1, 1, "", "Pathway"]], "pyseqrna.pathway.Pathway": [[3, 2, 1, "", "barplotKEGG"], [3, 2, 1, "", "dotplotKEGG"], [3, 2, 1, "", "enrichKEGG"]], "pyseqrna.pyseqrna_plots": [[3, 3, 1, "", "plotHeatmap"], [3, 3, 1, "", "plotMA"], [3, 3, 1, "", "plotVenn"], [3, 3, 1, "", "plotVolcano"]], "pyseqrna.pyseqrna_stats": [[3, 3, 1, "", "align_stats"]], "pyseqrna.pyseqrna_utils": [[3, 3, 1, "", "PyseqrnaLogger"], [3, 3, 1, "", "change_attribute"], [3, 3, 1, "", "change_ids"], [3, 3, 1, "", "check_files"], [3, 3, 1, "", "check_path"], [3, 3, 1, "", "check_status"], [3, 3, 1, "", "clusterRun"], [3, 3, 1, "", "findFiles"], [3, 3, 1, "", "getFiles"], [3, 3, 1, "", "getGenes"], [3, 3, 1, "", "get_basename"], [3, 3, 1, "", "get_cpu"], [3, 3, 1, "", "get_directory"], [3, 3, 1, "", "get_file_extension"], [3, 3, 1, "", "get_parent"], [3, 3, 1, "", "make_directory"], [3, 3, 1, "", "parse_config_file"], [3, 3, 1, "", "parse_gff"], [3, 3, 1, "", "read_input_file"], [3, 3, 1, "", "replace_cpu"]], "pyseqrna.quality_check": [[3, 3, 1, "", "fastqcRun"]], "pyseqrna.quality_trimming": [[3, 3, 1, "", "flexbarRun"], [3, 3, 1, "", "trim_galoreRun"], [3, 3, 1, "", "trimmomaticRun"]], "pyseqrna.quantification": [[3, 3, 1, "", "featureCount"], [3, 3, 1, "", "htseqCount"]], "pyseqrna.ribosomal": [[3, 3, 1, "", "sortmernaRun"]]}, "objtypes": {"0": "py:module", "1": "py:class", "2": "py:method", "3": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "class", "Python class"], "2": ["py", "method", "Python method"], "3": ["py", "function", "Python function"]}, "titleterms": {"pyseqrna": [0, 3], "an": 0, "autom": 0, "python": 0, "packag": 0, "rnaseq": 0, "data": 0, "analysi": 0, "introduct": [0, 3], "exampl": 0, "case": 0, "studi": 0, "content": 0, "indic": 0, "tabl": 0, "instal": 1, "modul": [2, 3], "qualiti": 3, "submodul": 3, "quality_check": 3, "quality_trim": 3, "ribosom": 3, "align": 3, "pyseqrna_stat": 3, "quantif": 3, "multimapped_group": 3, "normalize_count": 3, "cluster": 3, "differenti": 3, "gene": 3, "differential_express": 3, "function": 3, "annot": 3, "gene_ontologi": 3, "pathwai": 3, "util": 3, "arg_pars": 3, "pyseqrna_util": 3, "version": 3, "pyseqrna_plot": 3, "usag": 4}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 57}, "alltitles": {"pySeqRNA: an automated Python package for RNASeq data analysis": [[0, "pyseqrna-an-automated-python-package-for-rnaseq-data-analysis"]], "Introduction": [[0, "introduction"], [3, "module-pyseqrna"]], "Examples and Case-Studies": [[0, "examples-and-case-studies"]], "Contents": [[0, "contents"]], "Indices and tables": [[0, "indices-and-tables"]], "Installation": [[1, "installation"]], "Modules": [[2, "modules"]], "Quality Submodules": [[3, "quality-submodules"]], "pyseqrna.quality_check module": [[3, "module-pyseqrna.quality_check"]], "pyseqrna.quality_trimming module": [[3, "module-pyseqrna.quality_trimming"]], "pyseqrna.ribosomal module": [[3, "module-pyseqrna.ribosomal"]], "Alignment Submodules": [[3, "alignment-submodules"]], "pyseqrna.aligners module": [[3, "module-pyseqrna.aligners"]], "pyseqrna.pyseqrna_stats module": [[3, "module-pyseqrna.pyseqrna_stats"]], "Quantification Submodules": [[3, "quantification-submodules"]], "pyseqrna.quantification module": [[3, "module-pyseqrna.quantification"]], "pyseqrna.multimapped_groups module": [[3, "module-pyseqrna.multimapped_groups"]], "pyseqrna.normalize_counts module": [[3, "module-pyseqrna.normalize_counts"]], "pyseqrna.clustering module": [[3, "module-pyseqrna.clustering"]], "Differential Genes Submodules": [[3, "differential-genes-submodules"]], "pyseqrna.differential_expression module": [[3, "module-pyseqrna.differential_expression"]], "Functional Annotation Submodules": [[3, "functional-annotation-submodules"]], "pyseqrna.gene_ontology module": [[3, "module-pyseqrna.gene_ontology"]], "pyseqrna.pathway module": [[3, "module-pyseqrna.pathway"]], "Utility Submodules": [[3, "utility-submodules"]], "pyseqrna.arg_parser module": [[3, "module-pyseqrna.arg_parser"]], "pyseqrna.pyseqrna_utils module": [[3, "module-pyseqrna.pyseqrna_utils"]], "pyseqrna.version module": [[3, "module-pyseqrna.version"]], "pyseqrna.pyseqrna_plots module": [[3, "module-pyseqrna.pyseqrna_plots"]], "Usage": [[4, "usage"]]}, "indexentries": {"cpm() (pyseqrna.normalize_counts.normalization method)": [[3, "pyseqrna.normalize_counts.Normalization.CPM"]], "geneontology (class in pyseqrna.gene_ontology)": [[3, "pyseqrna.gene_ontology.GeneOntology"]], "gene_description (class in pyseqrna.differential_expression)": [[3, "pyseqrna.differential_expression.Gene_Description"]], "normalization (class in pyseqrna.normalize_counts)": [[3, "pyseqrna.normalize_counts.Normalization"]], "pathway (class in pyseqrna.pathway)": [[3, "pyseqrna.pathway.Pathway"]], "pyseqrnalogger() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.PyseqrnaLogger"]], "rpkm() (pyseqrna.normalize_counts.normalization method)": [[3, "pyseqrna.normalize_counts.Normalization.RPKM"]], "star_aligner (class in pyseqrna.aligners)": [[3, "pyseqrna.aligners.STAR_Aligner"]], "tpm() (pyseqrna.normalize_counts.normalization method)": [[3, "pyseqrna.normalize_counts.Normalization.TPM"]], "add_names() (pyseqrna.differential_expression.gene_description method)": [[3, "pyseqrna.differential_expression.Gene_Description.add_names"]], "add_names_annotation() (pyseqrna.differential_expression.gene_description method)": [[3, "pyseqrna.differential_expression.Gene_Description.add_names_annotation"]], "align_stats() (in module pyseqrna.pyseqrna_stats)": [[3, "pyseqrna.pyseqrna_stats.align_stats"]], "barplotgo() (pyseqrna.gene_ontology.geneontology method)": [[3, "pyseqrna.gene_ontology.GeneOntology.barplotGO"]], "barplotkegg() (pyseqrna.pathway.pathway method)": [[3, "pyseqrna.pathway.Pathway.barplotKEGG"]], "boxplot() (in module pyseqrna.normalize_counts)": [[3, "pyseqrna.normalize_counts.boxplot"]], "build_index() (pyseqrna.aligners.star_aligner method)": [[3, "pyseqrna.aligners.STAR_Aligner.build_index"]], "build_index() (pyseqrna.aligners.hisat2_aligner method)": [[3, "pyseqrna.aligners.hisat2_Aligner.build_index"]], "change_attribute() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.change_attribute"]], "change_ids() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.change_ids"]], "check_files() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.check_files"]], "check_index() (pyseqrna.aligners.star_aligner method)": [[3, "pyseqrna.aligners.STAR_Aligner.check_index"]], "check_index() (pyseqrna.aligners.hisat2_aligner method)": [[3, "pyseqrna.aligners.hisat2_Aligner.check_index"]], "check_path() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.check_path"]], "check_status() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.check_status"]], "clusterrun() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.clusterRun"]], "clustersample() (in module pyseqrna.clustering)": [[3, "pyseqrna.clustering.clusterSample"]], "countmmg() (in module pyseqrna.multimapped_groups)": [[3, "pyseqrna.multimapped_groups.countMMG"]], "degfilter() (in module pyseqrna.differential_expression)": [[3, "pyseqrna.differential_expression.degFilter"]], "dotplotgo() (pyseqrna.gene_ontology.geneontology method)": [[3, "pyseqrna.gene_ontology.GeneOntology.dotplotGO"]], "dotplotkegg() (pyseqrna.pathway.pathway method)": [[3, "pyseqrna.pathway.Pathway.dotplotKEGG"]], "enrichgo() (pyseqrna.gene_ontology.geneontology method)": [[3, "pyseqrna.gene_ontology.GeneOntology.enrichGO"]], "enrichkegg() (pyseqrna.pathway.pathway method)": [[3, "pyseqrna.pathway.Pathway.enrichKEGG"]], "fastqcrun() (in module pyseqrna.quality_check)": [[3, "pyseqrna.quality_check.fastqcRun"]], "featurecount() (in module pyseqrna.quantification)": [[3, "pyseqrna.quantification.featureCount"]], "findfiles() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.findFiles"]], "flexbarrun() (in module pyseqrna.quality_trimming)": [[3, "pyseqrna.quality_trimming.flexbarRun"]], "getfiles() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.getFiles"]], "getgenes() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.getGenes"]], "get_basename() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.get_basename"]], "get_cpu() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.get_cpu"]], "get_directory() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.get_directory"]], "get_file_extension() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.get_file_extension"]], "get_parent() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.get_parent"]], "hisat2_aligner (class in pyseqrna.aligners)": [[3, "pyseqrna.aligners.hisat2_Aligner"]], "htseqcount() (in module pyseqrna.quantification)": [[3, "pyseqrna.quantification.htseqCount"]], "leaf_label() (in module pyseqrna.clustering)": [[3, "pyseqrna.clustering.leaf_label"]], "make_directory() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.make_directory"]], "meanratiocount() (pyseqrna.normalize_counts.normalization method)": [[3, "pyseqrna.normalize_counts.Normalization.meanRatioCount"]], "module": [[3, "module-pyseqrna"], [3, "module-pyseqrna.aligners"], [3, "module-pyseqrna.arg_parser"], [3, "module-pyseqrna.clustering"], [3, "module-pyseqrna.differential_expression"], [3, "module-pyseqrna.gene_ontology"], [3, "module-pyseqrna.multimapped_groups"], [3, "module-pyseqrna.normalize_counts"], [3, "module-pyseqrna.pathway"], [3, "module-pyseqrna.pyseqrna_plots"], [3, "module-pyseqrna.pyseqrna_stats"], [3, "module-pyseqrna.pyseqrna_utils"], [3, "module-pyseqrna.quality_check"], [3, "module-pyseqrna.quality_trimming"], [3, "module-pyseqrna.quantification"], [3, "module-pyseqrna.ribosomal"], [3, "module-pyseqrna.version"]], "parse_config_file() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.parse_config_file"]], "parse_gff() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.parse_gff"]], "plotheatmap() (in module pyseqrna.pyseqrna_plots)": [[3, "pyseqrna.pyseqrna_plots.plotHeatmap"]], "plotma() (in module pyseqrna.pyseqrna_plots)": [[3, "pyseqrna.pyseqrna_plots.plotMA"]], "plotvenn() (in module pyseqrna.pyseqrna_plots)": [[3, "pyseqrna.pyseqrna_plots.plotVenn"]], "plotvolcano() (in module pyseqrna.pyseqrna_plots)": [[3, "pyseqrna.pyseqrna_plots.plotVolcano"]], "pyseqrna": [[3, "module-pyseqrna"]], "pyseqrna.aligners": [[3, "module-pyseqrna.aligners"]], "pyseqrna.arg_parser": [[3, "module-pyseqrna.arg_parser"]], "pyseqrna.clustering": [[3, "module-pyseqrna.clustering"]], "pyseqrna.differential_expression": [[3, "module-pyseqrna.differential_expression"]], "pyseqrna.gene_ontology": [[3, "module-pyseqrna.gene_ontology"]], "pyseqrna.multimapped_groups": [[3, "module-pyseqrna.multimapped_groups"]], "pyseqrna.normalize_counts": [[3, "module-pyseqrna.normalize_counts"]], "pyseqrna.pathway": [[3, "module-pyseqrna.pathway"]], "pyseqrna.pyseqrna_plots": [[3, "module-pyseqrna.pyseqrna_plots"]], "pyseqrna.pyseqrna_stats": [[3, "module-pyseqrna.pyseqrna_stats"]], "pyseqrna.pyseqrna_utils": [[3, "module-pyseqrna.pyseqrna_utils"]], "pyseqrna.quality_check": [[3, "module-pyseqrna.quality_check"]], "pyseqrna.quality_trimming": [[3, "module-pyseqrna.quality_trimming"]], "pyseqrna.quantification": [[3, "module-pyseqrna.quantification"]], "pyseqrna.ribosomal": [[3, "module-pyseqrna.ribosomal"]], "pyseqrna.version": [[3, "module-pyseqrna.version"]], "read_input_file() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.read_input_file"]], "replace_cpu() (in module pyseqrna.pyseqrna_utils)": [[3, "pyseqrna.pyseqrna_utils.replace_cpu"]], "rundeseq2() (in module pyseqrna.differential_expression)": [[3, "pyseqrna.differential_expression.runDESeq2"]], "run_alignment() (pyseqrna.aligners.star_aligner method)": [[3, "pyseqrna.aligners.STAR_Aligner.run_Alignment"]], "run_alignment() (pyseqrna.aligners.hisat2_aligner method)": [[3, "pyseqrna.aligners.hisat2_Aligner.run_Alignment"]], "run_edger() (in module pyseqrna.differential_expression)": [[3, "pyseqrna.differential_expression.run_edgeR"]], "sortmernarun() (in module pyseqrna.ribosomal)": [[3, "pyseqrna.ribosomal.sortmernaRun"]], "trim_galorerun() (in module pyseqrna.quality_trimming)": [[3, "pyseqrna.quality_trimming.trim_galoreRun"]], "trimmomaticrun() (in module pyseqrna.quality_trimming)": [[3, "pyseqrna.quality_trimming.trimmomaticRun"]]}})